src/box/tuple_extract_key.cc

```diff
@@ -254,21 +254,25 @@ tuple_extract_key_slowpath_raw(const char *data, const char *data_end,
 	assert(!key_def->is_multikey || multikey_idx != MULTIKEY_NONE);
 	assert(!key_def->for_func_index);
 	assert(mp_sizeof_nil() == 1);
-	/* allocate buffer with maximal possible size */
-	char *key = (char *) region_alloc(&fiber()->gc, data_end - data);
-	if (key == NULL) {
-		diag_set(OutOfMemory, data_end - data, "region",
-			 "tuple_extract_key_raw");
-		return NULL;
-	}
-	char *key_buf = mp_encode_array(key, key_def->part_count);
 	const char *field0 = data;
 	uint32_t field_count = mp_decode_array(&field0);
 	/*
 	 * A tuple can not be empty - at least a pk always exists.
 	 */
 	assert(field_count > 0);
 	(void) field_count;
+	/* Allocate buffer with maximal possible size. */
+	uint32_t potential_null_count = key_def->is_nullable ?
+		key_def->part_count : 0;
+	uint32_t alloc_size = (uint32_t)(data_end - data) +
+		potential_null_count * mp_sizeof_nil();
+	char *key = (char *)region_alloc(&fiber()->gc, alloc_size);
+	if (key == NULL) {
+		diag_set(OutOfMemory, alloc_size, "region",
+			 "tuple_extract_key_raw");
+		return NULL;
+	}
+	char *key_buf = mp_encode_array(key, key_def->part_count);
```

Comment:
Nit: Please always try to keep diff as possible, provided it doesn't hurt code readability. You don't need to move this hunk of code.

Reply:
Fixed